hedera-node/src/main/java/com/hedera/services/store/AccountStore.java

```diff
@@ -137,7 +137,10 @@ private Account loadEntityOrFailWith(Id id, @Nullable ResponseCodeEnum explicitR
 		final var merkleAccount = accounts.getImmutableRef(id.asGrpcAccount());
 
 		validateUsable(merkleAccount, explicitResponseCode, nonExistingCode, deletedCode);
+		return loadMerkleAccount(merkleAccount, id);
+	}
 
+	public static Account loadMerkleAccount(final MerkleAccount merkleAccount, final Id id) {
```

Comment:
This looks suspicious to me---by making this method public and using it directly in `AllowanceChecks`, we're skipping [several validations](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/store/AccountStore.java#L199) that were previously being done by `AccountStore`. (For example, `AllowanceChecks` will no longer fail on an expired owner account.)

I think we want `AllowanceChecks` to just use its `StateView.asReadOnlyBackingAccounts()` to construct an `AccountStore`---and then use that `AccountStore` as in the existing code. This way all the validations are preserved.

Reply:
Constructed accountStore and TypedTokenStore in the `AllowanceChecks` using `StateView` readonly backing stores. Removed the validations added in the AllowanceChecks to avoid duplication